src/__tests__/index.test.tsx

```diff
@@ -427,6 +432,24 @@ describe('gallery', () => {
     )
   })
 
+  test('should change location hash on slide change, only when gallery ID is set', async () => {
+    const user = userEvent.setup()
+    const items = createItems(1)
+
+    const { unmount } = render(<TestGallery items={items} id="my-gallery" />)
+    await user.click(screen.getAllByRole('img')[0])
+    dispatch('change')
+    expect(window.location.hash).toBe('#&gid=my-gallery&pid=1337')
+
+    closePhotoSwipe()
+    unmount()
```

Comment:
and add checking, that after photoswipe closing `location.hash` does not contain `gid` and `pid` params?

Reply:
added in [b60e480](https://github.com/dromru/react-photoswipe-gallery/pull/592/commits/b60e48002ac4a2a3b7e46d68e3131bc199632fc9)